packages/node-sdk/src/client.ts

```diff
@@ -0,0 +1,607 @@
+import { evaluateFlag } from "@bucketco/flag-evaluation";
+
+import cache from "./cache";
+import {
+  API_HOST,
+  BUCKET_LOG_PREFIX,
+  FLAG_EVENTS_PER_MIN,
+  FLAGS_REFETCH_MS,
+  SDK_VERSION,
+  SDK_VERSION_HEADER_NAME,
+} from "./config";
+import fetchClient from "./fetch-http-client";
+import {
+  Attributes,
+  Cache,
+  ClientOptions,
+  FeatureFlagEvent,
+  Flag,
+  FlagDefinitions,
+  Flags,
+  HttpClient,
+  Logger,
+  TrackOptions,
+} from "./types";
+import {
+  decorateLogger,
+  isObject,
+  ok,
+  rateLimited,
+  readNotifyProxy,
+} from "./utils";
+
+/**
+ * The SDK client.
+ *
+ * @remarks
+ * The client is used to interact with the Bucket API.
+ * Use the client to track users, companies, events, and feature flag usage.
+ **/
+export class BucketClient {
```

Comment:
If you call `trackFeatureUsage` on the client directly, we'll throw. Did you look at trying to handle this through types instead? Something like this could potentially work. The idea is instead of calling `new BucketBaseClient` directly, you'll call `const client = BucketClient(...)`. It will return an interface without `trackFeatureUsage` so you'll need to do `withUser` first to get "access" to that method:

```typescript
type BucketUserClient = {
  withCompany(
    companyId: string,
    opts?: { attributes?: Attributes },
  ): BucketCompanyClient;

  trackFeatureUsage(event: string, opts?: TrackOptions): Promise<boolean>;
  trackUser(opts?: TrackOptions): Promise<boolean>;
  withOtherContext(
    context: Record<string, any>,
    opts?: { replace?: boolean },
  ): BucketUserClient;
  otherContext: Record<string, any> | undefined;
  user: { userId: string; attrs?: Attributes } | undefined;
} & BucketBaseClient;

export type BucketCompanyClient = {
  company: { companyId: string; attrs?: Attributes } | undefined;
  trackCompany(opts?: TrackOptions): Promise<boolean>;
} & BucketUserClient;

interface BucketClientI {
  withUser(
    userId: string,
    opts?: { attributes?: Attributes },
  ): BucketUserClient;
  initialize(): Promise<void>;
  getFlags<TFlagKey extends string = string>(): Readonly<
    Record<TFlagKey, Flag>
  >;
}

/**
 * The SDK client.
 *
 * @remarks
 * The client is used to interact with the Bucket API.
 * Use the client to track users, companies, events, and feature flag usage.
 **/
function BucketClient(options: ClientOptions): BucketClientI {
  return new BucketBaseClient(options);
}

class BucketBaseClient {
  private _shared: {
    logger?: Logger;
    host: string;
    httpClient: HttpClient;
    refetchInterval: number;
    staleWarningInterval: number;
[...]
```